pkgs/development/libraries/openturns/default.nix

```diff
@@ -0,0 +1,82 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, swig
+, boost
+, spectra
+, libxml2
+, tbb
+, hmat-oss
+, nlopt
+, cminpack
+, ceres-solver
+, dlib
+, hdf5
+, primesieve
+, pagmo2
+, ipopt
+# tests take an hour to build on a 48-core machine
+, runTests ? false
+, enablePython ? false
+, python3Packages
+, darwin
+}:
+
+stdenv.mkDerivation rec {
+  pname = "openturns";
+  version = "1.19";
+
+  src = fetchFromGitHub {
+    owner = "openturns";
+    repo = "openturns";
+    rev = "v1.19";
+    sha256 = "sha256-hSvhKx7Ke5PXPYGcn9OJvq7YooAbHrbP6TdLZBotriE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [
+    swig
+    boost
+    spectra
+    libxml2
+    tbb
+    hmat-oss
+    nlopt
+    cminpack
+    ceres-solver
+    dlib
+    hdf5
+    primesieve
+    pagmo2
+    ipopt
+  ] ++ lib.optionals enablePython [
+    python3Packages.python
+    python3Packages.matplotlib
+    python3Packages.psutil
+    python3Packages.dill
+  ] ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Accelerate;
+
+  cmakeFlags = [
+    "-DOPENTURNS_SYSCONFIG_PATH=$out/etc"
+    "-DUSE_SPHINX=OFF"
```

Comment:
Is it possible to enable Sphinx to get documentation, maybe by using `sphinxBuildHook` if it makes sense?

Reply:
I've enabled the DUSE_SPHINX flag. sphinxHook doesn't seem to work out of the box since it's not letting cmake configure Sphinx's config.py before it runs, so all it sees is config.py.in rather than config.py (trying to search for config.py in nix build directory generated with the -K flag gave nothing). According to [this file](https://github.com/openturns/openturns/blob/master/python/src/CMakeLists.txt) SPHINX_BIN_DIR should be generated somewhere...